include/itkAttenuationImageFilter.hxx

```diff
@@ -0,0 +1,225 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkAttenuationImageFilter_hxx
+#define itkAttenuationImageFilter_hxx
+
+#include "itk_eigen.h"
+#include "itkMath.h"
+#include "itkImageLinearConstIteratorWithIndex.h"
+#include "itkImageSink.h"
+#include "itkImageRegionSplitterDirection.h"
+#include "itkSimpleDataObjectDecorator.h"
+
+namespace itk
+{
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::AttenuationImageFilter()
+{
+  this->SetNumberOfRequiredInputs(1);
+  this->DynamicMultiThreadingOn();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+const ImageRegionSplitterBase *
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::GetImageRegionSplitter() const
+{
+  m_RegionSplitter->SetDirection(m_Direction);
+  return m_RegionSplitter.GetPointer();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::BeforeThreadedGenerateData()
+{
+  // Verify inputs
+  if (this->GetMaskImage() == nullptr)
+  {
+    itkExceptionMacro("Filter requires a mask image for inclusion estimates!");
+  }
+
+  if (this->GetDirection() > ImageDimension)
+  {
+    itkExceptionMacro("Scan line direction must be a valid image dimension!");
+  }
+
+  if (this->GetSamplingFrequencyMHz() < itk::Math::eps)
+  {
+    itkExceptionMacro("RF sampling frequency was not set!");
+  }
+
+  Superclass::BeforeThreadedGenerateData();
+
+  // Initialize metric image
+  this->GetOutput()->Allocate();
+  this->GetOutput()->FillBuffer(0.0f);
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::AfterThreadedGenerateData()
+{
+  Superclass::AfterThreadedGenerateData();
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::DynamicThreadedGenerateData(
+  const OutputRegionType & regionForThread)
+{
+  if (regionForThread.GetNumberOfPixels() == 0)
+  {
+    return;
+  }
+
+  const MaskImageType *  maskImage = this->GetMaskImage();
+  const InputImageType * input = this->GetInput();
+  OutputImageType *      output = this->GetOutput();
+
+  ImageLinearConstIteratorWithIndex<TInputImage> it(input, regionForThread);
+  it.SetDirection(m_Direction);
+  it.GoToBegin();
+
+  unsigned int   inclusionLength;
+  InputIndexType start, end;
+
+  const float scanStepMM = input->GetSpacing()[m_Direction];
+
```

Comment:
Add this here:
```cpp
  const IndexValueType fixedDistance = m_FixedEstimationDepth / scanStepMM;
```

Reply:
For consistency with padding I have opted to store pixel distance in `m_FixedEstimationDepth` and expose a setter that transforms from spatial (mm) distance to pixel distance.